app/src/main/java/com/example/drawingactivity/ColorWheel.kt

```diff
@@ -0,0 +1,60 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.util.AttributeSet
+import android.util.Log
+import android.view.View
+
+@SuppressLint("ClickableViewAccessibility")
+class ColorWheel @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    var changeColor: ((Int) -> Unit)? = null
+
+    var wheelColor = Color.BLACK
+    var index = 0
```

Comment:
If I'm understanding you correctly, I think you want a function that's public that advances the color forward in the list of colors

Reply:
now included in viewModel